src/Signature/SignatureV4.php

```diff
@@ -76,6 +112,26 @@ public function signRequest(
         return $this->buildRequest($parsed);
     }
 
+    /**
+     * Get the headers that were used to pre-sign the request.
+     * Used for the X-Amz-SignedHeaders header.
+     *
+     * @param array $headers
+     * @return array
+     */
+    private function getPresignHeaders(array $headers)
+    {
+        $presignHeaders = [];
+        $blacklist = $this->getHeaderBlacklist();
+        foreach ($headers as $name => $value) {
+            $lName = strtolower($name);
+            if (!isset($blacklist[$lName]) && $name !== self::AMZ_CONTENT_SHA256_HEADER) {
```

Comment:
`getPresignHeaders` should to also exclude the headers that match `stripos($name, 'x-amz-') !== 0)` to prevent bloating presigned URLs with the headers that don't need to be there - See my `'Note on S3'`.

Reply:
Can you also please update the formatting for this line, since it's over 80 characters?

```php
if (!isset($blacklist[$lName])
    && $name !== self::AMZ_CONTENT_SHA256_HEADER
    && stripos($name, 'x-amz-') !== 0
) {
    $presignHeaders[] = $lName;
}
```